routing/datasource.go

```diff
@@ -482,6 +482,10 @@ func processRouteDef(cpm map[string]PredicateSpec, fr filters.Registry, def *esk
 		return nil, err
 	}
 
+	scheme, host, err = NormalizeSchemeHost(scheme, host)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This can be moved inside of splitBackend.

Then it looks like that all three call sites first parse a string via url.Parse or url.ParseRequestURI. Reading the docs https://pkg.go.dev/net/url#ParseRequestURI I think ParseRequestURI is the right one to use since we expect absolute URI.

So I propose that we create a helper like
```go
func SchemeHost(u string) (scheme, host string, error)
```
and use it everywhere.

Reply:
Please also add a godoc